### PR TITLE
TypeScript support for the existing Node.js example

### PR DESCRIPTION
**Pull request by onpaws**
Hi there, I'm Pat. First time contributor. Thanks for your efforts on skaffold!

Fixes: #4261

### Description
The [example Node.js project](https://github.com/GoogleContainerTools/skaffold/tree/master/examples/nodejs) is a great starting point for JavaScript developers. Some developers may prefer to use TypeScript, so this PR intends to do the minimum number of steps to port the existing Node.js example to TypeScript, making sure hot reload still works in the expected way.

#### Steps I took:
- Duplicated `examples/nodejs` and renamed to `examples/typescript`
- `npm install typescript`
- `tsc --init` and accepted all defaults.
- Added the minimum necessary type dependencies (`@types/express` and `@types/node`) to keep the compiler happy.
- Ported the two JavaScript files to TypeScript.
- Swapped [nodemon](https://nodemon.io/) for [tsc-watch](https://github.com/gilamran/tsc-watch#the-nodemon-for-typescript), which appears to be drop-in compatible.

**Follow-up Work**
[VS Code] Potentially consider [remote debugging](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack)

#### Tests
I didn't write new tests because I'm not sure what exactly it should mean to "test" an example.

That being said `make test` seems to still pass:
<summary>
<details>

```
Generated proto files are updated!
PASSED hack/test-generated-proto.sh in 2s
Running linters...
RUN hack/boilerplate.sh
PASSED hack/boilerplate.sh in 1s
RUN hack/gofmt.sh
PASSED hack/gofmt.sh in 1s
RUN hack/golangci-lint.sh
Installing GolangCI-Lint
golangci/golangci-lint info checking GitHub for tag 'v1.30.0'
golangci/golangci-lint info found version: 1.30.0 for v1.30.0/darwin/amd64
golangci/golangci-lint info installed /Users/paws/Developer/skaffold/hack/bin/golangci-lint
PASSED hack/golangci-lint.sh in 62s
```

</details>
</summary>

Same for `make integration` -- the new `examples/typescript` folder did show up in the output and it appears to pass also. (I'm not going to pretend I understand what this test suite is actually doing :) Maybe it copies all the examples into `examples/`?)

```
<snip>
=== RUN   TestDiagnose/typescript
time="2021-01-31T13:51:49+01:00" level=info msg="Running [skaffold diagnose] in examples/typescript"
time="2021-01-31T13:51:49+01:00" level=info msg="Ran [skaffold diagnose] in 107.417913ms"
<snip>
--- PASS: TestDiagnose/typescript (0.11s)
<snip>
```

#### Versions
```
OS: macOS 11.1 20C69
Node: v14.15.4
TypeScript: 4.1.3
```

### CODE REVIEW

**Comment by onpaws** [comment]
@googlebot I fixed it.

**Comment by onpaws** [comment]
I see Travis tests mostly pass, except for `buildpacks`, which I find a bit curious b/c I didn't touch anything to do with buildpacks, at least AFAIK. Happy to look into it - have a feeling rebase-ing this PR off some other commit than `master` may resolve. Is there a better branch to target? What do you think?

`FAIL: TestDebug/buildpacks (390.59s)`

**Comment by IsaacPD** [comment]
Thanks for the contribution @onpaws , I believe there's an issue with the `buildpacks` test, I've been seeing failures on some of our recent PRs. So I would not worry to much about it but feel free to report any findings related to the issue to #5326 .

In regards to the changes you made, you added the example under `integration\examples`, could you also duplicate it under `examples`?

**Comment by gsquared94** [comment]
> In regards to the changes you made, you added the example under `integration\examples`, could you also duplicate it under `examples`?

@IsaacPD the `examples` directory gets populated from `integration/exampels` on every new version release. It doesn't need to be done manually.

File: integration/examples/typescript/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+### Example: TypeScript + Node.js with hot-reload
+
+Seeks to be functionally identical to the [Node.js](./nodejs) example, except with TypeScript.
```
Comment (gsquared94) [inline integration/examples/typescript/README.md:3]: should be `[Node.js](../nodejs)` ?

File: integration/examples/typescript/skaffold.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        context: backend
+        docker:
+          buildArgs:
+            ENV: development
+        sync:
+          manual:
+          # Sync all the TypeScript files that are in the src folder
+          # with the container src folder
+          - src: 'src/**/*.ts'
+            dest: .
```
Comment (gsquared94) [inline integration/examples/typescript/skaffold.yaml:25]:
```suggestion
            dest: .

```

File: integration/examples/typescript/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+#### Workflow
+
+* Make some changes to [index.ts](./backend/src/index.ts):
+    * The file will be synchronized to the cluster
+    * `tsc-watch` will restart the application
+* Make some changes to `package.json`:
+    * The full build/push/deploy process will be triggered, fetching dependencies from `npm`
+
+
```
Comment (gsquared94) [inline integration/examples/typescript/README.md:21]:
```suggestion

```

File: integration/examples/typescript/README.md (hunk 1 above)
Comment (onpaws) [reply integration/examples/typescript/README.md:3]: Indeed, thanks

**Review by gsquared94 — APPROVED** [review]

**Comment by onpaws** [comment]
@IsaacPD my first commit actually put it in examples/ like you said, but then I saw the README [section](https://github.com/GoogleContainerTools/skaffold/blob/master/examples/README.md) 'Note for contributors' and it sounds like a build script does it automatically. Thanks for looking out :)

**Comment by IsaacPD** [comment]
The test have that are failing have been temporarily disabled on master, could you please rebase to catch these changes and prevent travis from failing. Thanks

**Comment by onpaws** [comment]
Yep, you got it, coming right up

**Comment by onpaws** [comment]
Hopefully this is the desired commit:
`097325d65..c254d2512  master          -> upstream/master`

Let's see if the tests pass this time 🤞

**Pull request merged**